lib/absinthe/schema/notation.ex

```diff
@@ -1139,29 +1139,30 @@ defmodule Absinthe.Schema.Notation do
   """
   defmacro import_types(type_module_ast) do
     env = __CALLER__
-    {:ok, _} =
-      type_module_ast
-      |> Macro.expand(env)
-      |> do_import_types(env)
+    type_module_ast
+    |> Macro.expand(env)
+    |> do_import_types(env)
     :ok
   end
 
   defp do_import_types({{:., _, [root_ast, :{}]}, _, modules_ast_list}, env) do
-    # root_ast = {:__aliases__, [alias: false], [:MyApp, :Schema, :Types]}
-    # modules_ast_list = [
-    #    {:__aliases__, [alias: false], [:TypesA]},
-    #    {:__aliases__, [alias: false], [:TypesB]}
-    #  ]
-    {_, _, root} = root_ast
-    Enum.each(modules_ast_list, fn {_, _, leaf} ->
-      do_import_types( root ++ leaf |> Enum.join(".") |> String.to_atom , env)
-    end)
-  end
+    {:__aliases__, _, root} = root_ast
 
+    root_module = Module.concat(root)
+    root_module_with_alias = Keyword.get(env.aliases, root_module, root_module)
+
+    for {_, _, leaf} <- modules_ast_list do
+      type_module = Module.concat([root_module_with_alias | leaf])
+      if Code.ensure_loaded?(type_module) do
```

Comment:
@ssomnoremac What happens if you remove this `ensure_loaded?` check in the project where you're seeing errors?

Reply:
@bruce I can't reliably reproduce the error unfortunately, but I'll try that and work with it all week and let you know if I get the error again, manually recompiling as much as possible.

Reply:
Thanks; let me know if you find a clear reproducible case. I'll try to circle back when I have time later this week.

Reply:
@bruce , been using this for well over a week now and haven't had any issues. Deleted the `_build` folder as well to check that. Hasn't failed on me yet.

Reply:
Had you tried deleting `_build` before? I wonder if things got into a weird state somehow.

Reply:
no, someone suggested I do that to test that removing `ensure_loaded?` wouldn't affect the initial build. But before, no, I wasn't deleting the folder.